src/journal/journalctl.c

```diff
@@ -2420,10 +2428,21 @@ int main(int argc, char *argv[]) {
                 }
         }
 
+        if (arg_cursor_walk) {
+                _cleanup_free_ char *cur = NULL;
+                if (0 == read_one_line_file(arg_cursor_walk, &cur)
+                    && 0 <= sd_journal_seek_cursor(j, cur)) {
+                        arg_after_cursor = cur;
+                        cur = NULL;
```

Comment:
TAKE_PTR() is your friend. Even better: free_and_replace()

this leaks memory if `--after-cursor=` is used here too. Combining both switches should really result in an error early on I figure..

Reply:
memory management of arg_after_cursor is a bit chaotic. The variable should either always point to heap memory, or always into argv[] but not sometimes one and sometimes the other. Otherwise it's really messy it properly free it at the end, since heap memory needs that and argv[] memory doesn't.